syntax = "proto3";

option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package example;

service BiddingService {
  rpc PlaceBid (BidRequest) returns (BidReply) {}
  rpc GetBidHistory (BidHistoryRequest) returns (BidHistoryReply) {}
  rpc GetHighestBid (HighestBidRequest) returns (HighestBidReply) {}
}

message BidRequest {
  int32 auction_id = 1;
  int32 bidder_id = 2;
  double price = 3;
}

message BidReply {
  string message = 1;
}

message BidHistoryRequest {
  int32 auction_id_filter = 1;
  int32 next_bid_id = 2;
  int32 limit = 3;
}

message Bid {
  int32 bid_id = 1;
  int32 auction_id = 2;
  int32 bidder_id = 3;
  double bid_price = 4;
  string bid_at = 5;
}

message BidHistoryReply {
  repeated Bid bids = 1;
  int32 next_bid_id = 2;
}

message HighestBidRequest {
  int32 auction_id_filter = 1;
}

message HighestBidReply {
  repeated Bid bids = 1;
}
